src/overlays/actors/ovl_En_Test2/z_en_test2.c

```diff
@@ -26,22 +39,105 @@ const ActorInit En_Test2_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A3E74C[] = {
+EnTest2DisplayObjs sDisplayObjects[] = {
+    { object_dekucity_ana_obj_DL_000040, NULL, NULL },
+    { object_sichitai_obj_DL_001820, NULL, NULL },
+    { object_yukimura_obj_DL_0008C0, NULL, NULL },
+    { object_hakugin_obj_DL_0016D8, NULL, object_hakugin_obj_Matanimheader_0017A8 },
+    { object_hakugin_obj_DL_002018, NULL, object_hakugin_obj_Matanimheader_0020E8 },
+    { object_hakugin_obj_DL_005268, NULL, object_hakugin_obj_Matanimheader_005338 },
+    { object_meganeana_obj_DL_000110, object_meganeana_obj_DL_000080, NULL },
+    { object_haka_obj_DL_000F70, NULL, NULL },
+    { object_haka_obj_DL_001200, NULL, NULL },
+    { object_hakugin_obj_DL_004928, NULL, object_hakugin_obj_Matanimheader_0049F8 },
+    { object_hakugin_obj_DL_002978, NULL, object_hakugin_obj_Matanimheader_002A58 },
+    { object_hakugin_obj_DL_000D38, NULL, object_hakugin_obj_Matanimheader_000E48 },
+    { object_hakugin_obj_DL_0035A8, NULL, object_hakugin_obj_Matanimheader_003888 },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 8000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 2500, ICHAIN_STOP),
 };
 
-#endif
+static s16 sTest2Objects[] = {
+    OBJECT_DEKUCITY_ANA_OBJ, OBJECT_SICHITAI_OBJ,  OBJECT_YUKIMURA_OBJ, OBJECT_HAKUGIN_OBJ, OBJECT_HAKUGIN_OBJ,
+    OBJECT_HAKUGIN_OBJ,      OBJECT_MEGANEANA_OBJ, OBJECT_HAKA_OBJ,     OBJECT_HAKA_OBJ,    OBJECT_HAKUGIN_OBJ,
+    OBJECT_HAKUGIN_OBJ,      OBJECT_HAKUGIN_OBJ,   OBJECT_HAKUGIN_OBJ,
+};
+
+void EnTest2_Init(Actor* thisx, PlayState* play) {
+    EnTest2* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if ((this->actor.params == 0xB) || (this->actor.params == 0xC)) {
+        this->actor.flags |= ACTOR_FLAG_20;
+    }
+}
+
+void EnTest2_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    s32 objectIndex;
+    EnTest2DisplayObjs* dispObjects;
+    EnTest2* this = THIS;
+    objectIndex = Object_GetIndex(&play->objectCtx, sTest2Objects[this->actor.params]);
+
+    if (objectIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    if (Object_IsLoaded(&play->objectCtx, objectIndex)) {
+        dispObjects = &sDisplayObjects[this->actor.params];
+        this->actor.objBankIndex = objectIndex;
+        this->actor.draw = EnTest2_Draw;
+        if ((dispObjects->animMat) != 0) {
```

Comment:
```suggestion
        if (dispObjects->animMat != NULL) {
```
Pointer, compare to NULL.